toucan_connectors/pagination.py

```diff
@@ -0,0 +1,88 @@
+from typing import Literal
+
+from pydantic import BaseModel
+
+
+class OffsetLimitInfo(BaseModel):
+    # Both of these must be positive integers
+    offset: int
+    # Limit can be None because charts need to be able to retrieve all data
+    limit: int | None
+
+
+class UnknownSizeDatasetPaginationInfo(BaseModel):
+    type: Literal['unknown_size'] = 'unknown_size'
+    # Will be true if there is no more information to fetch
+    is_last_page: bool
+
+
+class KnownSizeDatasetPaginationInfo(BaseModel):
+    type: Literal['known_size'] = 'known_size'
+    # Will be true if there is no more information to fetch
+    is_last_page: bool
+    # the total size of the dataset
+    total_rows: int
+
+
+class PaginationInfo(BaseModel):
+    # A recap of the provided parameters
+    parameters: OffsetLimitInfo
+    pagination_info: UnknownSizeDatasetPaginationInfo | KnownSizeDatasetPaginationInfo
```

Comment:
IMO to prevent having `pagination_info.pagination_info` everywhere, it could have been great to rename this attribute as `pagintation_type`, so that, we could access with `DataSlice.pagination_info.pagination_type`